frontend/src/pages/modelRegistry/screens/utils.ts

```diff
@@ -115,23 +117,29 @@ export const filterRegisteredModels = (
   unfilteredRegisteredModels: RegisteredModel[],
   search: string,
   searchType: SearchType,
-): RegisteredModel[] =>
-  unfilteredRegisteredModels.filter((rm: RegisteredModel) => {
+): RegisteredModel[] => {
+  const searchLower = search.toLowerCase();
+
+  return unfilteredRegisteredModels.filter((rm: RegisteredModel) => {
     if (!search) {
       return true;
     }
 
     switch (searchType) {
-      case SearchType.KEYWORD:
+      case SearchType.KEYWORD: {
         return (
-          rm.name.toLowerCase().includes(search.toLowerCase()) ||
-          (rm.description && rm.description.toLowerCase().includes(search.toLowerCase()))
+          rm.name.toLowerCase().includes(searchLower) ||
+          (rm.description && rm.description.toLowerCase().includes(searchLower)) ||
+          getLabels(rm.customProperties).some((label) => label.toLowerCase().includes(searchLower))
         );
+      }
 
-      case SearchType.OWNER:
-        return rm.owner && rm.owner.toLowerCase().includes(search.toLowerCase());
+      case SearchType.OWNER: {
+        return rm.owner && rm.owner.toLowerCase().includes(searchLower);
```

Comment:
![image](https://github.com/user-attachments/assets/4bacd777-7541-4ebd-94f0-3465735c7624)

Seems you're missing a test?

Reply:
It's interesting but it's probably missing a test even before that because I applied a small change to those lines( and not really added them from scratch). Now I'm trying to cover owner search, but we don't have owner anywhere in our types, only in author. So I wonder if this functionality is even valid in the first place? @andrewballantyne

Reply:
haha -- interesting -- I know very little about MR, so if the code is never run -- we probably should trim it?

Reply:
@YuliaKrimerman registered models have owners while model versions have authors.

Reply:
i offered a suggestion about merging or filtering out some of the common code between the two at the end of this comment  https://github.com/opendatahub-io/odh-dashboard/pull/3288#discussion_r1785535731

Reply:
The versions tests were added, and I replied on that other comment, I think we can follow up on that refactor in another PR